xarray/plot/dataset_plot.py

```diff
@@ -49,7 +49,7 @@ def _infer_meta_data(ds, x, y, hue, hue_style, add_guide, funcname):
             add_colorbar = False
             add_legend = False
     else:
-        if add_guide is True and funcname != "quiver":
+        if add_guide is True and not (funcname == "quiver" or funcname == "streamplot"):
```

Comment:
it might be easier to use `in` for this:
```suggestion
        if add_guide is True and funcname not in ("quiver", "streamplot"):
```